### ISSUE TITLE
Mismatch of Kwinana Cogen facility capacity in OpenNEM with other sources (such as the owner's website)

### ISSUE THREAD

**Issue by aleith** [opennem/opennem#104]
Kwinana Cogeneration Plant registered capacity
[OpenNEM](https://opennem.org.au/facility/au/WEM/TIWEST_COG1/?range=all&interval=fin-year) : 45 MW
[Engie](https://engie.com.au/home/assets/gas/kwinana)/Wikipedia: 120 MW

In addition to seeing big mismatches between electricity production data (#100) in OpenNEM and the reporting to the Clean Energy Regulator under the Safeguard Mechanism, I'm seeing some numbers on plant capacity the are confusing. 

Often the registered capacity is a couple of percent different in OpenNEM to other sources, but in some cases it's much more different. the case of Kwinana Cogeneration Plant is less than half what is on Engie's own website. In this case it may have something to do with the fact that they produced power for the WEM and the BP refinery it is colocated with, and that is behind the meter sales, but it should all be registered capacity, I think.

**Comment by aleith** [comment]
another one,
Parkston Power Station register capacity:
[OpenNEM](https://opennem.org.au/facility/au/WEM/PRK_AG/?range=all&interval=fin-year): 68.0 MW
[Owner](https://www.transaltarenewables.com/plants-operation/parkeston/): 110 MW
This week [in the media](https://www.kalminer.com.au/news/regional/parkeston-power-plant-a-viable-alternative-for-backup-power-honey-says-c-7606196): 110 MW

**Comment by aleith** [comment]
Kambalda possibly wrong,
OpenNEM: 23.0 MW
WIkipedia: 42.0 MW
hopefully somebody at [Transalta in Kalgoorlie](https://www.whitepages.com.au/trans-alta-energy-10293402/kalgoorlie-wa-10293400B) will call me today and let me know the capacity and the capacities of all the other power stations in Kal. 

The facility names in OpenNEM are not consistent with the registered facility names in the CER database, certainly for all the FF generation in WA that I've been examining closely —  so it's become quite confusing trying to sort all this out.

**Comment by nc9** [comment]
> The facility names in OpenNEM are not consistent with the registered facility names in the CER database, certainly for many of the FF generation facilities in WA that I've been examining closely

Hi @aleith - we store both the original network name and the "clean" OpenNEM name - if you have suggestions on what needs to be cleaned up here just tag us and let us know. 

Combined with the open issue on emission factors in the WEM it would be good to go through and check all the data

All of the stations and units are now in a single JSON file - this gets deployed automatically to our dev instances (https://dev.opennem.org.au) with every push/build/pull request 

It should be accessible enough now for external edits and pull requests

https://github.com/opennem/opennem/blob/master/opennem/data/stations.json

**Comment by aleith** [comment]
Hi @Nic — I'll add some coal closured date pull requests today to the JSON file you linked to. Still working on consolidating the three data sources for facility info and electricity and emissions into a table or two for easy filtering and sorting to help get to the bottom of the differences for the WA facilities (someone else can take on the NEM!).

I downloaded the SCADA Facility names table and the generation data for FY21 from WA AEMO and I'm slowly compiling a table of comparative data from SCADA, OpenNEM, and the CEG Safeguard Mechanism reporting.

Some of the names are really, really confusing, especially when you have four or more facilities in one location for e.g. Kalgoorlie or Kwinana and the CEG data doesn't report facility capacity so hard to differentiate sometimes.

Significant differences already, for e.g. the four diesel generators owned by Tesla Inc (not that @Telsa) are in OpenNEM but only two of them are in the Safeguard Mech reporting (they never run except for testing and receive payments under the RCM that WA has). There's capacity differences b/w the SCADA and OpenNEM and info on the web, like owners websites, reporting of sales of facilities etc. Once it's all done I'll post the table, I already have a table of differences in the FY21 electricity and emissions reporting from OpenNEM and CER reporting at #100. As you can see there, the differences are often small, 2-5% but they do run as high as 76%(!)

**Comment by nc9** [comment]
We should probably merge #100 into here and just have a general issue for facility data quality including capacities, close dates and emissions.

There is a closure data parser in the repo but it requires joining the data based on facility name - which is a bit of a manual process

**Comment by aleith** [comment]
It's only 5 units that have been scheduled for closure in the NEM so far. I can do it manually in the JSON. The dates the Govt provided are in the "Oct 2022" format. Should I make them "15 Oct 2022" or "15/Oct/2022" or what?

Also, I think it would be helpful to add an ownership field to the JSON, because it can help disambiguate facilities and help front end OpenNEM users get some clues as to which facility is which, there's a bunch of gas turbines around the place because of the RCM paying them to be around and also the gas crisis a few years ago when a rusty [Varanas Island pipeline blew up](https://www.boilingcold.com.au/the-varanus-island-gas-explosion-that-shook-the-wa-economy/).

    "code": "MUJA",
    "name": "Muja CD",
    "network_name": "Muja CD",
    "description": null,
    "wikipedia_link": null,
    "wikidata_id": null,
    "website_url": null,
    "approved": true,
    "approved_by": "opennem.registry",
add: 
    "owner_clean": "Synergy"
    "owner_code": "ELECTRICITY GENERATION AND RETAIL CORPORATION".   <— from Clean Energy Regulator[ Safeguard Mechanism database.](https://www.cleanenergyregulator.gov.au/NGER/The-safeguard-mechanism/safeguard-data)

**Comment by aleith** [comment]
one other thing I'm noticing is that the CER database and info on the web indicates that many of the facilities listed in OpenNEM as `"fueltech_id": "gas_ocgt"`, are burning diesel or `"distillate"` as it's referred to in OpenNEM. These are multi-fuel gas turbines. Also it's likely that all new gas/diesel generation will be reciprocating engine type because it has a good combination of gas ramping, two maintenance and decent efficiency compared to OCGT, more like CCGT efficiencies.

**Comment by nc9** [comment]
> or "15/Oct/2022" or what?

ISO 8601 format - the database has closure date down to a full datetime, so it would be:

`2022-10-22T00:00:00+10:00`

In python to check:

```python
In [17]: datetime.fromisoformat("2022-10-22T00:00:00+10:00")
Out[17]: datetime.datetime(2022, 10, 22, 0, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=36000)))
```

**Comment by nc9** [comment]
I'm wondering if we want to track the difference between _expected_ closure date, and _actual_ closure date @simonhac

**Comment by aleith** [comment]
ok so given they are just saying Oct 2022 should I make it 1st, 15th or 31st Oct 2022 for  Muja C unit G5?

**Comment by nc9** [comment]
If the specificity of the date is only down to a year or month would default to the first of any year/month - it's a major pain otherwise to store truncated values

**Comment by aleith** [comment]
yep, that's why i suggested 15th. for example the WA Govt said Collie Coal will close in 2027 in public but then i you go to their media release it says Oct 2027, which lets it run through winter, the most profitable period for coal. If we assumed Jan 1 2027 we'd have been a year early. 

My guess is Synergy will run Collie out to 31 Oct 2027 if they can :-)

**Comment by simonhac** [comment]
> If the specificity of the date is only down to a year or month would default to the first of any year/month - it's a major pain otherwise to store truncated values

i've often wanted a non-specific date type that allows YYYY, YYYY-MM, YYYY-MM-DD, YYYY-MM-DDTHH, YYYY-MM-DDTHH:MM and YYYY-MM-DDTHH:MM:SS etc.

eg. we know that liddell started in 1971, but we can be quite sure that it didn't start at 1970-01-01-00:00:00+10, so we don't want to default to the wrong day, month, HMS.

**Comment by simonhac** [comment]
this is how [global energy observer](http://globalenergyobservatory.org/form.php?pid=43145) does it:

<img width="949" alt="image" src="https://user-images.githubusercontent.com/20500649/183350266-9598f577-a0d7-4056-8e85-eed57ab09067.png">

**Comment by nc9** [comment]
> i've often wanted a non-specific date type that allows YYYY, YYYY-MM, YYYY-MM-DD, YYYY-MM-DDTHH, YYYY-MM-DDTHH:MM and YYYY-MM-DDTHH:MM:SS etc.

This is a problem in a lot of projects. Way I previously solved it was to store a full timestamp and then have the specificity in a separate field (ie. `year`, `month`, `day`) and then when you query it out you do `date_trunc(specificity, datetime)`

**Comment by simonhac** [comment]
would be great if there was a type that had a full timestamp and a resolution parameter… eg. 1 means just year, 2 means year+month, 3 means year+month+day etc.

**Comment by aleith** [comment]
thats one way to solve it. i'm guessing most of the codebase assumes ISO 8601 compliant time stamps and so data conversion / type coercion (yuk) needs to happen at least for this to work. 

something around date types needs to happen however it's solved i suppose.

**Comment by nc9** [comment]
> > i've often wanted a non-specific date type that allows YYYY, YYYY-MM, YYYY-MM-DD, YYYY-MM-DDTHH, YYYY-MM-DDTHH:MM and YYYY-MM-DDTHH:MM:SS etc.
> 
> This is a problem in a lot of projects. Way I previously solved it was to store a full timestamp and then have the specificity in a separate field (ie. `year`, `month`, `day`) and then when you query it out you do `date_trunc(specificity, datetime)`

copy @chienleng here since I'll be setting this out in the API shortly. you should be able to date_trunc using date-fns

**Comment by nc9** [comment]
#151 related - I have a schema now on dev to store the history here as well

**Comment by nc9** [comment]
#151
